### ISSUE TITLE
x/build: consider opting-in to longtest builders by default for cmd/go changes, if/when they're under 15 min

### ISSUE THREAD

**Issue by dmitshur** [golang/go#42661]
(I have a feeling @bcmills may have already filed an issue for this, but I couldn't find it. Maybe it was a comment instead.)

With the progress made on https://github.com/golang/go/issues/37439#issuecomment-723413000 recently, it seems we can speed up longtest SlowBots to run in under 15 minutes or so. If that can be done, what is your preference on enabling longtest builders by default for changes that touch the `go` command? CC @bcmills, @jayconrod, @matloob.

In either case, we need to wait for #37439 to be completed first, and give it a little time to see if that on its own is sufficient. But I wanted to create a tracking issue to start gathering feedback eagerly.

I'm also looking for feedback on:

- If this is something we want to do, would you prefer that "cmd/go" changes are determined by the commit subject having the "cmd/go" path present (which is up to the author), or by looking whether any files in src/cmd/go directory are modified?

Also CC @golang/release.

**Comment by bcmills** [comment]
> (I have a feeling @bcmills may have already filed an issue for this, but I couldn't find it. Maybe it was a comment instead.)

I think it was previously an implicit subtext of #37439. Thanks for filing it as a proper issue!

**Comment by bcmills** [comment]
If the longtest builders can be made to run at roughly the same latency as other TryBots, I think we would prefer to have it trigger by default (in much the same way that the `x/tools` 'bot triggers automatically).

I think we should use the same mechanism that we currently use to trigger `x/tools`, whatever that is. (I can't find it at the moment...)

**Comment by matloob** [comment]
This would be great! It would save me from accidentally submitting bugs in many cases...

**Comment by bcmills** [comment]
We should also run `longtest` builders by default any time there are changes in `src/vendor` or `src/cmd/vendor`. We've had too many build breaks lately due to stray edits.

**Comment by mknyszek** [comment]
Which of these features is still desirable? This has become pretty easy to do with LUCI and I already sent out a CL to add run linux-amd64-staticlockranking when a CL modifies runtime files so the context is all fresh.

**Comment by dmitshur** [comment]
I agree with the non-cmd/go-related suggestion in https://github.com/golang/go/issues/42661#issuecomment-1116274332. That is, I suggest opting into longtest on at least linux/amd64 when any of these files are modified:

- src/{,cmd/}go.mod
- src/{,cmd/}vendor/*
- src/*_bundle.go

I defer to @bcmills for cmd/go details. Thanks @mknyszek!

**Comment by mknyszek** [comment]
I sent out a CL for what you listed @dmitshur.

@bcmills That CL only seems to cover vendoring. Are there any other Go command related file paths that should enable the longtest builder? Thanks.
